src/resolvers/githubresolver.ts

```diff
@@ -1,30 +1,47 @@
 import Debug from "debug";
-import * as GitInfo from "hosted-git-info";
 import { SubResolver } from ".";
 
 const debug = Debug("resolverengine:githubresolver");
 
-// hosted-git-info is a godsend, but it doesn't support specific files
-// use a hack to capture third slash a file skipping commitish
-// 1st group - protocol, location, owner, repo
-// 2nd group - file inside repo
-// 3rd group - comittish
-const FILE_LOCATION_REGEX = /^((?:.+:\/\/)?[^:/]+[/:][^/]+[/][^/]+)[/](.+?)(#.+)?$/;
+const BROWSER_LINK_REGEX = /^https?:\/\/github\.com\/([^/]+)\/([^/]+)\/blob\/((?:[^/]+[/])*[^/]+)$/;
+const GITHUB_URI_REGEX = /^github:([^/]+)\/([^/]+)\/((?:[^/#]+[/])*[^/#]+?)(#\w+)?$/;
+const WEIRD_BROWSER_LINK_REGEX = /^https?:\/\/github\.com\/([^/]+)\/([^/]+)\/((?:[^/]+[/])*[^/]+)$/; // remix
 
 // TODO(ritave): Support private repositories
 export function GithubResolver(): SubResolver {
   return async (what: string): Promise<string | null> => {
-    const fileMatch = what.match(FILE_LOCATION_REGEX);
-    if (!fileMatch) {
-      return null;
+    // GitInfo returned invalid results (ask me about this: %2F)
+
+    const fileMatchLink = what.match(BROWSER_LINK_REGEX);
+    if (fileMatchLink) {
+      const [, owner, repo, commitAndFile] = fileMatchLink;
+      const gitRawUrl = `https://raw.githubusercontent.com/${owner}/${repo}/${commitAndFile}`;
```

Comment:
Duplication of work - hosted-git-info resolves into raw urls - https://github.com/npm/hosted-git-info#methods

> For example `hostedGitInfo.fromUrl("git@github.com:npm/hosted-git-info.git#v1.0.0").file("package.json")` would return `https://raw.githubusercontet.com/npm/hosted-git-info/v1.0.0/package.json`